compile.js

```diff
@@ -15,28 +15,53 @@ const BytesPath = path.resolve(__dirname, 'contracts', 'BytesLib.sol')
 const SafeMathPath = path.resolve(__dirname, 'contracts', 'SafeMath.sol')
 const SigCheckPath = path.resolve(__dirname, 'contracts', 'SigCheck.sol')
 
-
-let input = {
-    'SPVStore.sol': fs.readFileSync(SPVStorePath, 'utf8'),
-    'BTCUtils.sol': fs.readFileSync(BTCUtilsPath, 'utf8'),
-    'ValidateSPV.sol': fs.readFileSync(ValidateSPVPath, 'utf8'),
-    'BytesLib.sol': fs.readFileSync(BytesPath, 'utf8'),
-    'SafeMath.sol': fs.readFileSync(SafeMathPath, 'utf8'),
-    'SigCheck.sol': fs.readFileSync(SigCheckPath, 'utf8')
+let sources = {
+    'SPVStore.sol': {
+        content: fs.readFileSync(SPVStorePath, 'utf8')
+    },
+    'BTCUtils.sol': {
+        content: fs.readFileSync(BTCUtilsPath, 'utf8')
+    },
+    'ValidateSPV.sol': {
+        content: fs.readFileSync(ValidateSPVPath, 'utf8')
+    },
+    'BytesLib.sol': {
+        content: fs.readFileSync(BytesPath, 'utf8')
+    },
+    'SafeMath.sol': {
+        content: fs.readFileSync(SafeMathPath, 'utf8')
+    },
+    'SigCheck.sol': {
+        content: fs.readFileSync(SigCheckPath, 'utf8')
+    }
 }
 
-const output = solc.compile({sources: input}, 1);
+var input = {
+    language: 'Solidity',
+    sources: sources,
+    settings: {
+        outputSelection: {
+            '*': {
+                '*': ['*'],
+            },
+        },
+    },
+};
+
+const output = JSON.parse(solc.compile(JSON.stringify(input)))
 
 // log errors
 if (output.errors) {
-    console.log(output.errors);
+    console.error(output.errors);
 }
 
 // Save compiled contracts to json files
 for (let contract in output.contracts) {
-    contract_name = contract.split(':');
+    let fileName = contract.split('.sol')[0];
+    let contractName = Object.keys(output.contracts[contract])[0]
```

Comment:
We have to separate `fileName` and `contractName` variables because the contract name doesn't have to match the file name. 
We have such this situation for `SigCheck.sol` which defines `CheckBitcoinSigs` library.